plugins/filter/lib/filterlist.cpp

```diff
@@ -61,7 +64,53 @@ void FilterList::set(const QStringList &list)
     d->m_conf.set(list);
 }
 
-bool FilterList::match(const QString &number)
+void FilterList::removeEntry(const QString &entry)
+{
+    QStringList filters = list();
+    if (filters.removeAll(entry) > 0) {
+        set(filters);
+    }
+}
+
+void FilterList::addEntry(const QString &entry)
+{
+    QStringList filters = list();
+    if (!filters.contains(entry)) {
+        filters.prepend(entry);
+        set(filters);
+    }
+}
+
+void FilterList::clear()
+{
+    set(QStringList());
+}
+
+bool FilterList::match(const QString &number) const
+{
+    for (const QString &filter : list()) {
+        if (filter.startsWith('+') || filter[0].isDigit()) {
+            // Exact number matching
+            if (number == filter
+                || CommHistory::remoteAddressMatch(CommHistory::RING_ACCOUNT, number, filter, true)) {
```

Comment:
Uh, this libcommhistory part is icky. First the "commonutils.h" is somewhat dumping ground for random thing, but then this RING_ACCOUNT is just ugly. Moreover the QStringLiteral() there will now create extra instances wherever the header is included.

What if we at least add something like phoneNumberMatch() which would do the same thing but not requiring to pass ring account? And the RING_ACCOUNT we could move away from the public api, I didn't spot any usage outside libcommhistory at the moment. I can also do that.

Reply:
> What if we at least add something like phoneNumberMatch() which would do the same thing but not requiring to pass ring account? And the RING_ACCOUNT we could move away from the public api, I didn't spot any usage outside libcommhistory at the moment. I can also do that.

I totally agree. If you prefer to concentrate on other tasks, tell me. I can do it also.